genesis/camino_genesis_test.go

```diff
@@ -0,0 +1,290 @@
+					thisConfig.Camino.Allocations = append(thisConfig.Camino.Allocations, KopernikusConfig.Camino.Allocations...)
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("wrong validator duration"),
+		},
+		"repeated staker allocation": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.Allocations = []CaminoAllocation{{AddressStates: AddressStates{ConsortiumMember: true, KYCVerified: true}, PlatformAllocations: []PlatformAllocation{{NodeID: KopernikusConfig.Camino.Allocations[0].PlatformAllocations[0].NodeID, ValidatorDuration: 1}}}}
+					thisConfig.Camino.Allocations = append(thisConfig.Camino.Allocations, KopernikusConfig.Camino.Allocations...)
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("repeated staker allocation"),
+		},
+		"no staker allocations": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.Allocations = []CaminoAllocation{{PlatformAllocations: []PlatformAllocation{}}}
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("no staker allocations"),
+		},
+		"valid": {
+			args: args{
+				config: &KopernikusConfig,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := validateCaminoConfig(tt.args.config)
+
+			if tt.err != nil {
+				require.ErrorContains(t, err, tt.err.Error())
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestBuildCaminoGenesis(t *testing.T) {
+	ctx := snow.DefaultContextTest()
+	rewards := reward.NewCalculator(defaultRewardConfig)
+	ctx.NetworkID = constants.KopernikusID
+	baseDB := memdb.New()
+	db := versiondb.New(baseDB)
+	cfg := defaultConfig()
+
+	addrs := set.Set[ids.ShortID]{}
+	_, _, avaxAddrBytes, _ := address.Parse(initialAdmin)
+	avaxAddr, _ := ids.ToShortID(avaxAddrBytes)
+	addrs.Add(avaxAddr)
+	outputOwners := secp256k1fx.OutputOwners{
+		Locktime:  0,
+		Threshold: 1,
+		Addrs:     []ids.ShortID{avaxAddr},
+	}
+	type args struct {
+		config *Config
+		hrp    string
+	}
+	tests := map[string]struct {
+		args          args
+		expectedUtxos map[ids.ID]*avax.UTXO
+		err           error
+	}{
+		"success - kopernikus": {
+			args: args{
+				config: &KopernikusConfig,
+				hrp:    constants.KopernikusHRP,
+			},
+			expectedUtxos: map[ids.ID]*avax.UTXO{
+				expectedUtxoID1: generateTestUTXO(expectedUtxoID1, ctx.AVAXAssetID, 2000000000000, outputOwners, expectedUtxoID1, expectedBondTxID),
+				expectedUtxoID2: generateTestUTXO(expectedUtxoID2, ctx.AVAXAssetID, 10000000000000000, outputOwners, expectedUtxoID2, ids.Empty),
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			genesisBytes, _, err := buildCaminoGenesis(tt.args.config, tt.args.hrp)
+			if tt.err != nil {
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+			s := defaultState(genesisBytes, cfg, ctx, db, rewards)
+
+			utxos, err := avax.GetAllUTXOs(s, addrs)
```

Comment:
I'm curious, will this also work for x-chain utxos? It should, right? And we need to check all possible allocation cases including x-chain allocations